sdk/python/feast/infra/utils/aws_utils.py

```diff
@@ -406,3 +406,89 @@ def unload_redshift_query_to_df(
         drop_columns,
     )
     return table.to_pandas()
+
+
+def get_lambda_function(lambda_client, function_name: str) -> Optional[Dict]:
+    """
+    Get the AWS Lambda function by name or return None if it doesn't exist.
+    Args:
+        lambda_client: AWS Lambda client.
+        function_name: Name of the AWS Lambda function.
+
+    Returns: Either a dictionary containing the get_function API response, or None if it doesn't exist.
+
+    """
+    try:
+        return lambda_client.get_function(FunctionName=function_name)["Configuration"]
+    except ClientError as ce:
+        # If the resource could not be found, return False.
+        # Otherwise bubble up the exception (most likely permission errors)
+        if ce.response["Error"]["Code"] == "ResourceNotFoundException":
+            return None
+        else:
+            raise
+
+
+def delete_lambda_function(lambda_client, function_name: str) -> Dict:
+    """
+    Delete the AWS Lambda function by name.
+    Args:
+        lambda_client: AWS Lambda client.
+        function_name: Name of the AWS Lambda function.
+
+    Returns: The delete_function API response dict
+
+    """
+    return lambda_client.delete_function(FunctionName=function_name)
+
+
+def get_first_api_gateway(api_gateway_client, api_gateway_name: str) -> Optional[Dict]:
+    """
+    Get the first API Gateway with the given name. Note, that API Gateways can have the same name.
+    They are identified by AWS-generated ID, which is unique. Therefore this method lists all API
+    Gateways and returns the first one with matching name. If no matching name is found, None is returned.
+    Args:
+        api_gateway_client: API Gateway V2 Client.
+        api_gateway_name: Name of the API Gateway function.
+
+    Returns: Either a dictionary containing the get_api response, or None if it doesn't exist
+
+    """
+    response = api_gateway_client.get_apis()
+    apis = response.get("Items", [])
+
+    while True:
```

Comment:
We should probably limit how often we page through this API response (something sufficiently high like 10 or something).